drivers/include/periph/spi.h

```diff
@@ -130,104 +143,87 @@ int spi_init_master(spi_t dev, spi_conf_t conf, spi_speed_t speed);
  * @param[in] cb        callback called every time a byte was received
  *
  * @return              0 on success
- * @return              -1 on error
+ * @return              -1 on undefined device given
+ * @return              -2 on other errors
  */
 int spi_init_slave(spi_t dev, spi_conf_t conf, char (*cb)(char data));
 
 /**
- * @brief Configure SCK, MISO and MOSI pins for the given SPI device
- *
- * @param[in] dev       SPI device to use
+ * @brief   Configure SCK, MISO and MOSI pins for the given SPI device
  *
- * @return              0 on success
- * @return              -1 on error
+ * @param[in] dev       SPI device to configure pins for
  */
-int spi_conf_pins(spi_t dev);
+void spi_conf_pins(spi_t dev);
 
 /**
- * @brief Get mutually exclusive access to the given SPI bus
+ * @brief   Get mutually exclusive access to the given SPI bus
  *
- * In case the SPI device is busy, this function will block until the bus is free again.
+ * In case the SPI device is busy, this function will block until the bus is
+ * free again.
  *
  * @param[in] dev       SPI device to access
- *
- * @return              0 on success
- * @return              -1 on error
  */
-int spi_acquire(spi_t dev);
+void spi_acquire(spi_t dev);
 
 /**
- * @brief Release the given SPI device to be used by others
+ * @brief   Release the given SPI device to be used by others
  *
  * @param[in] dev       SPI device to release
- *
- * @return              0 on success
- * @return              -1 on error
  */
-int spi_release(spi_t dev);
+void spi_release(spi_t dev);
 
 /**
- * @brief Transfer one byte on the given SPI bus
+ * @brief   Transfer one byte on the given SPI bus
  *
  * @param[in] dev       SPI device to use
  * @param[in] out       Byte to send out, set NULL if only receiving
  * @param[out] in       Byte to read, set NULL if only sending
- *
- * @return              Number of bytes that were transfered
- * @return              -1 on error
  */
-int spi_transfer_byte(spi_t dev, char out, char *in);
+void spi_transfer_byte(spi_t dev, char out, char *in);
```

Comment:
I would prefer to have in as the return value instead of a pointer. If
nothing else it will allow the compiler to reduce stack usage by placing
the value in a register instead of RAM.